contracts/auctions/src/contract.rs

```diff
@@ -0,0 +1,433 @@
+    }
+
+    #[allow(dead_code)]
+    pub fn buy_now(env: Env, auction_id: u64, buyer: Address) -> Result<(), ContractError> {
+        buyer.require_auth();
+
+        let mut auction = get_auction_by_id(&env, auction_id)?;
+
+        if env.ledger().timestamp() > auction.end_time || auction.status != AuctionStatus::Active {
+            log!(&env, "Auction: Buy Now: Auction not active: ", auction_id);
+            return Err(ContractError::AuctionNotActive);
+        }
+
+        if auction.item_info.buy_now_price.is_none() {
+            log!(
+                env,
+                "Auction: Buy Now: trying to buy an item that does not allow `buy now`"
+            );
+            return Err(ContractError::NoBuyNowOption);
+        }
+
+        let old_highest_bid = get_highest_bid(&env, auction_id)?;
+
+        // refund any previous highest bid
+        let token = token::Client::new(&env, &auction.currency);
+
+        token.transfer(
+            &env.current_contract_address(),
+            &old_highest_bid.bidder,
+            &(old_highest_bid.bid as i128),
+        );
+
+        // pay for the item
+        token.transfer(
+            &buyer,
+            &auction.seller,
+            &(auction
+                .item_info
+                .buy_now_price
+                .expect("Auction: Buy Now: Buy now price has not been set") as i128),
+        );
+
+        let collection_client = collection::Client::new(&env, &auction.item_info.collection_addr);
+
+        collection_client.safe_transfer_from(
+            &env.current_contract_address(),
+            &auction.seller,
+            &buyer,
+            &auction.item_info.item_id,
+            &1,
+        );
+
+        auction.status = AuctionStatus::Ended;
+        auction.highest_bid = Some(
+            auction
+                .item_info
+                .buy_now_price
+                .expect("Auction: Buy Now: Buy now price has not been set"),
+        );
+
+        save_auction(&env, &auction)?;
+
+        env.events()
+            .publish(("buy now", "auction id: "), auction_id);
+        env.events().publish(("buy now", "buyer: "), buyer);
+
+        Ok(())
+    }
+
+    #[allow(dead_code)]
+    pub fn pause(env: Env, auction_id: u64) -> Result<(), ContractError> {
+        let mut auction = get_auction_by_id(&env, auction_id)?;
+        auction.seller.require_auth();
+
+        if auction.status != AuctionStatus::Active {
+            log!(
+                &env,
+                "Auction: Pause: Cannot pause inactive/ended auction: ",
+                auction_id
+            );
+            return Err(ContractError::AuctionNotActive);
+        }
+
+        if env.ledger().timestamp() > auction.end_time {
+            log!(&env, "Auction: Pause: Auction expired: ", auction_id);
+            return Err(ContractError::AuctionNotActive);
+        }
+
+        auction.status = AuctionStatus::Paused;
+
+        save_auction_by_id(&env, auction_id, &auction)?;
```

Comment:
Use helpers you have implemented.

```suggestion
        save_auction(&env, auction_id, &auction)?;
```